sdk/database/helper/connutil/sql_test.go

```
@@ -0,0 +1,56 @@
+package connutil
+
+import (
+	"context"
+	"net/url"
+	"testing"
+)
+
+func TestSQLPasswordChars(t *testing.T) {
+	testCases := []struct {
+		Username string
+		Password string
+	}{
+		{"postgres", "password{0}"},
+		{"postgres", "pass:word"},
+		{"postgres", "pass/word"},
+		{"postgres", "p@ssword"},
+		{"postgres", "pass\"word\""},
+		// Much to my surprise, CREATE USER "{{password}}" PASSWORD 'foo' worked.
+		{"{{password}}", "foo"},
+		{"user", "{{username}}"},
+	}
+	for _, tc := range testCases {
+		t.Logf("username %q password %q", tc.Username, tc.Password)
+
+		sql := &SQLConnectionProducer{}
+		ctx := context.Background()
+		conf := map[string]interface{}{
+			"connection_url": "postgres://{{username}}:{{password}}@localhost:5432/mydb",
+			"username":       tc.Username,
+			"password":       tc.Password,
+		}
+		_, err := sql.Init(ctx, conf, false)
+		if err != nil {
+			t.Errorf("Init error on %q %q: %+v", tc.Username, tc.Password, err)
+		} else {
+			// This jumps down a few layers...
+			// Connection() uses sql.Open uses lib/pq uses net/url.Parse
+			u, err := url.Parse(sql.ConnectionURL)
+			if err != nil {
+				t.Errorf("URL parse error on %q %q: %+v", tc.Username, tc.Password, err)
```

Can we change these to `t.Fatalf()` to fail fast?